[nn] add `insert` method to sequential class

khushi-411 opened the pull request:
Follows #71329

cc @kshitij12345

khushi-411 committed:
add insert method to seq class

khushi-411 commented:
Hi, gentle ping @albanD  @jbschlosser! :)

Thanks in advance!

kshitij12345 commented:
Hi @khushi-411,

@albanD @jbschlosser are on leave, so it might take a few days to get this PR reviewed.

kshitij12345 commented on torch/nn/modules/container.py:
```diff
@@ -163,6 +163,12 @@ def append(self, module: Module) -> 'Sequential':
         return self
 
+    def insert(self, index: int, module: Module) -> 'Sequential':
+        for i in range(len(self._modules), index, -1):
```
Also should we have an assert to verify that `module` is actually an instance of `Module`?

khushi-411 replied:
I think it will be great. Will add it. Thanks for suggesting it.

khushi-411 commented:
> Hi @khushi-411,
> 
> @albanD @jbschlosser are on leave, so it might take a few days to get this PR reviewed.

Hi, @kshitij12345!

Nice, thanks for sharing with me ;)

kshitij12345 reviewed (commented):
Overall looks good but I have a few inline questions.

Also, I was wondering if it should support negative indices. Python lists do support that.

```python
>>> a = [1, 2, 3]
>>> a.insert(-1, 0)
>>> a
[1, 2, 0, 3]
>>> a.insert(-2, 5)
>>> a
[1, 2, 5, 0, 3]
```

Wdyt?

khushi-411 committed:
add neg indices support and range check for index val

khushi-411 committed:
add assert to verify module is an instance of Module

khushi-411 committed:
Merge branch 'pytorch:master' into seq/insert

khushi-411 commented:
Thanks for reviewing, @kshitij12345!

> Wdyt?

I think it's a pretty cool suggestion. Thanks! I've added support for negative inputs and addressed all your reviews ;)

khushi-411 committed:
Merge branch 'master' into seq/insert

khushi-411 commented:
Oops! Resolving conflicts caused many testing errors, which looks reasonable to me. I'll take a look into those. Thanks!

khushi-411 committed:
add return statement

albanD reviewed (approved):
Looks good! Thanks for taking care of that!
